src/Table/index.jsx

```diff
@@ -1,16 +1,110 @@
 import React from 'react';
 import classNames from 'classnames';
+import FontAwesomeStyles from 'font-awesome/css/font-awesome.min.css';
+import isRequiredIf from 'react-proptype-conditional-require';
 import PropTypes from 'prop-types';
 
 import styles from './Table.scss';
+import Button from '../Button';
 
 class Table extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = props.tableSortable ?
+      {
+        sortedColumn: this.props.defaultSortedColumn,
+        sortDirection: this.props.defaultSortDirection,
+      }
+      :
+      {
+        sortedColumn: '',
+        sortDirection: '',
+      };
+
+    this.onSortClick = this.onSortClick.bind(this);
+  }
+
+  onSortClick(columnKey) {
+    let newDirection = 'desc';
+
+    if (this.state.sortedColumn === columnKey) {
+      newDirection = (this.state.sortDirection === 'desc' ? 'asc' : 'desc');
+
+      this.setState({
+        sortDirection: newDirection,
+      });
+    } else {
+      this.setState({
+        sortedColumn: columnKey,
+        sortDirection: newDirection,
+      });
+    }
+
+    const currentlySortedColumn = this.props.columns.find(
+      column => (columnKey === column.key));
+    currentlySortedColumn.onSort(newDirection);
+  }
+
   getCaption() {
     return this.props.caption && (
       <caption>{this.props.caption}</caption>
     );
   }
 
+  getHeadingText(columnKey) {
```

Comment:
This gets the sr-only text for the heading. I'm thinking this function name could be more clear.

Reply:
Maybe something like `getA11yHeadingText`?

Reply:
That works for me. I think I'll wait and see whether the table needs any additional accessibility work first, and rename things after, once I know they're actually covering accessibility cases.